datastore/google/cloud/datastore/query.py

```diff
@@ -86,7 +86,7 @@ def __init__(self,
         self._client = client
         self._kind = kind
         self._project = project or client.project
```

Comment:
Rather than propagating an AttributeError, please add a check that at least one of `client` or `project` is passed, and raise a ValueError with a useful error if not.